src/BotPlutusInterface/Config.hs

```diff
@@ -0,0 +1,281 @@
+
+  pcProtocolParamsFile <-
+    sectionWithDefault'
+      (pcProtocolParamsFile def)
+      "protocolParamsFile"
+      filepathSpec
+      "Protocol params file location relative to the cardano-cli working directory (needed for the cli) in JSON format. "
+
+  pcDryRun <-
+    sectionWithDefault'
+      (pcDryRun def)
+      "dryRun"
+      trueOrFalseSpec
+      "Dry run mode will build the tx, but skip the submit step"
+
+  pcLogLevel <-
+    sectionWithDefault' (pcLogLevel def) "logLevel" logLevelSpec ""
+
+  pcOwnPubKeyHash <-
+    sectionWithDefault (pcOwnPubKeyHash def) "ownPubKeyHash" ""
+
+  pcOwnStakePubKeyHash <-
+    sectionWithDefault' (pcOwnStakePubKeyHash def) "ownStakePubKeyHash" (maybeSpec anySpec) ""
+
+  pcTipPollingInterval <-
+    sectionWithDefault' (pcTipPollingInterval def) "tipPollingInterval" naturalSpec ""
+
+  pcPort <-
+    sectionWithDefault' (pcPort def) "port" portSpec ""
+
+  pcEnableTxEndpoint <-
+    sectionWithDefault' (pcEnableTxEndpoint def) "enableTxEndpoint" trueOrFalseSpec ""
+
+  pcCollectStats <-
+    sectionWithDefault'
+      (pcCollectStats def)
+      "collectStats"
+      trueOrFalseSpec
+      "Save some stats during contract run (only transactions execution budgets supported atm)"
+
+  pure PABConfig {..}
+
+docPABConfig :: String
+docPABConfig = show $ generateDocs pabConfigSpec
+
+{- |Load 'PABConfig' from the file:
+
+NOTE:
+
+- default value for: @pcProtocolParamsFile@ is "./protocol.json"
+
+- if @pcProtocolParamsFile == "./protocol.json"@ and file don't exist, try to
+  fetch in from @cardano-cli@ on the fly (only for local)
+
+- in other cases -- fail
+-}
+loadPABConfig :: FilePath -> IO (Either String PABConfig)
+loadPABConfig fn = do
+  confE <- deserialize <$> readFile fn
+  case confE of
+    Left err -> return $ Left $ "PABConfig: " <> fn <> ": " <> err
+    Right conf@PABConfig {pcProtocolParamsFile, pcNetwork, pcCliLocation} -> do
+      pparamsE <- readProtocolParametersJSON (toString pcProtocolParamsFile)
+      case pparamsE of
+        Left err
+          | pcProtocolParamsFile == "./protocol.json"
+              && pcCliLocation == Local -> do
+            let shellArgs =
+                  ShellArgs
+                    { cmdName = "cardano-cli"
+                    , cmdArgs =
+                        [ "query"
+                        , "protocol-parameters"
+                        , "--out-file"
+                        , "./protocol.json"
+                        ]
+                          ++ networkArg pcNetwork
+                    , cmdOutParser = id
+                    }
+            callLocalCommand shellArgs
+              >>= \case
+                Left errPParams -> return $ Left $ Text.unpack errPParams
+                Right _ -> loadPABConfig fn
+          | otherwise ->
+            return $ pparamsError pcProtocolParamsFile err
+        Right pcProtocolParams -> return $ Right conf {pcProtocolParams}
```

Comment:
Created a new issue for protocol params handling. I think with Misha's node querying work, we could just query for the params each time (or store in memory): https://github.com/mlabs-haskell/bot-plutus-interface/issues/112